src/Concordium/Client/GRPC2.hs

```diff
@@ -2359,42 +2360,343 @@ instance ToProto SendBlockItemInput where
             UpdateInstruction uInstruction ->
                 ProtoFields.updateInstruction .= toProto uInstruction
 
+-- |Information about a block which arrived at the node.
+data ArrivedBlockInfo = ArrivedBlockInfo {
+    -- |Hash of the block.
+    abiBlockHash :: !BlockHash,
+    -- |Absolute height of the block, where 0 is the height of the genesis block.
+    abiBlockHeight :: !AbsoluteBlockHeight
+} deriving (Show)
+
+instance FromProto Proto.ArrivedBlockInfo where
+    type Output' Proto.ArrivedBlockInfo = ArrivedBlockInfo
+    fromProto abInfo = do
+        abiBlockHash <- fromProto =<< abInfo ^? ProtoFields.hash
+        abiBlockHeight <- fromProto =<< abInfo ^? ProtoFields.height
+        return ArrivedBlockInfo{..}
+
+instance FromProto Proto.FinalizedBlockInfo where
+    type Output' Proto.FinalizedBlockInfo = ArrivedBlockInfo
+    fromProto abInfo = do
+        abiBlockHash <- fromProto =<< abInfo ^? ProtoFields.hash
+        abiBlockHeight <- fromProto =<< abInfo ^? ProtoFields.height
+        return ArrivedBlockInfo{..}
+
+instance FromProto Proto.InstanceStateKVPair where
+    type Output' Proto.InstanceStateKVPair = (ByteString, ByteString)
+    fromProto ikvPair = do
+        key <- ikvPair ^? ProtoFields.key
+        value <- ikvPair ^? ProtoFields.value
+        return (key, value)
+
+instance FromProto Proto.DelegatorInfo where
+    type Output' Proto.DelegatorInfo = QueryTypes.DelegatorInfo
+    fromProto dInfo = do
+        pdiAccount <- fromProto =<< dInfo ^? ProtoFields.account
+        pdiStake <- fromProto =<< dInfo ^? ProtoFields.stake
```

Comment:
So the same comments about using `^?` for lenses apply here as well as on the other PR. Please address those.